src/dmd/common/outbuffer.h

```diff
@@ -49,7 +49,7 @@ struct OutBuffer
     void reserve(d_size_t nbytes);
     void setsize(d_size_t size);
     void reset();
-    void write(const void *data, size_t nbytes);
+    void write(const void *data, d_size_t nbytes);
```

Comment:
The argument here needs to be the *host* size_t, not the target's.

Reply:
Alright; I'll have to give this more of a think over the next few days. Since as-is it's causing linker errors when building LDC on 32-bit OpenBSD. The classic `can't find signature matching Outbuffer::write(const void *, unsigned long), did you mean Outbuffer::write(const void *, unsigned int) ?`

Reply:
This is correct, `d_size_t` is the host size_t. I think Walter was confused about this patch being to compile with DMD to 32-bit, rather than that you need it to build DMD (LDC) itself on a 32-bit platform.

Reply:
Correct. In case it's not clear: I am on 32-bit OpenBSD and want to build 32-bit LDC. I am getting these errors with this setup.